### PR TITLE
fix(datetime): add aria-labels to interactive buttons

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Screen readers do not announce to show/hide the year picker when selecting the month/year toggle button. 

Issue URL: Internal

## What is the new behavior?

- Aria label has been added to the month/year toggle of the calendar for announcing "Show/Hide year picker" along with the current month/year
  - This aligns with how screen readers announce the iOS date picker
- Aria labels have been capitalized to align with other labels
- Screen readers announce show/hide year picker along with the current month/year

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by sean-perkins** [comment]
Note to self: Just need to add padding that existed on the `ion-item`, to the new container.

File: core/src/components/datetime/datetime.scss [hunk 1]
```diff
@@ -204,6 +204,45 @@ ion-picker-column-internal {
 // Calendar / Header / Action Buttons
 // -----------------------------------
 
+.calendar-month-year-toggle {
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime.scss:207]: Focus states are missing from this button

File: core/src/components/datetime/datetime.scss [hunk 2]
```diff
@@ -204,6 +204,45 @@ ion-picker-column-internal {
...
+  @include font-smoothing();
+  @include margin(0);
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime.scss:209]: A layout shift happens on MD when clicking the button

https://user-images.githubusercontent.com/2721089/216689222-f8e19d05-6d40-40a6-b51a-195dcc5b4ddc.mov

Comment (liamdebeasi) [reply core/src/components/datetime/datetime.scss:209]: Also, the ripple effect is missing on MD

File: core/src/components/datetime/datetime.scss [hunk 3]
```diff
@@ -204,6 +204,45 @@ ion-picker-column-internal {
...
+  @include padding(0px, 0px, 0px, 20px);
+  @include text-inherit();
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime.scss:211]: Hover states are missing too. However, they are missing on iOS mode in `main` today

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

**Comment by amandaejohnston** [comment]
What are all the extra style changes meant for here? Judging from the PR title/description, this should just be changing the `aria-label` behavior, but it looks like there are other things going on.

**Comment by sean-perkins** [comment]
@amandaejohnston this PR was accounting for too much scope, so I discussed offline with Liam and decided on splitting the work effort into two parts.

Part 1 is available here: https://github.com/ionic-team/ionic-framework/pull/26744
Part 2 is what I discussed in stand-up, just adding the `aria-label` to the button (once we are ok with the refactor changes).

**Comment by sean-perkins** [comment]
Closing in favor of https://github.com/ionic-team/ionic-framework/pull/26848

**Pull request closed without merging**